src/main/java/frc/robot/subsystems/arm/Arm.java

```diff
@@ -1,9 +1,135 @@
 package frc.robot.subsystems.arm;
 
+import java.util.TreeMap;
+
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.Follower;
+import com.ctre.phoenix6.controls.PositionVoltage;
+import com.ctre.phoenix6.hardware.CANcoder;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+
+import edu.wpi.first.util.sendable.SendableBuilder;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
-import frc.robot.Constants;
 import frc.robot.RobotMap;
 
 public class Arm extends SubsystemBase {
+    public final TalonFX armMotorLeft = new TalonFX(RobotMap.ARM_MOTOR_LEFT_CAN_ID);
+    public final TalonFX armMotorRight = new TalonFX(RobotMap.ARM_MOTOR_RIGHT_CAN_ID);
+
+    private final CANcoder armLefttEncoderAbs = new CANcoder(RobotMap.ARM_ABS_ENCODER_CAN_ID);
+    public enum ArmPosition {
+        // home will also be the position for when we are deep climbing
+        HOME,
+        INTAKE,
+        L2,
+        L3,
+        L4,
+    }
+
+    private double armDemandVolts = 0.0;
+    private double desiredSetpointPosition = 0.0;
+    private double desiredSetpointVelocity = 0.0;
+
+    /**
+     * Map each of our arm positions to an actual position on our arm (degrees)
+     */
+    public final TreeMap<ArmPosition, Double> armPositions = new TreeMap<ArmPosition, Double>();
+
+    private ArmPosition armDesiredPosition = ArmPosition.HOME;
+    public Arm() {
+        armPositions.put(ArmPosition.HOME, ArmCal.ARM_POSITION_HOME_DEGREES);
+        armPositions.put(ArmPosition.INTAKE, ArmCal.ARM_POSITION_INTAKE_DEGREES);
+        armPositions.put(ArmPosition.L2, ArmCal.ARM_POSITION_L2_DEGREES);
+        armPositions.put(ArmPosition.L3, ArmCal.ARM_POSITION_L3_DEGREES);
+        armPositions.put(ArmPosition.L4, ArmCal.ARM_POSITION_L4_DEGREES);
+
+        initArmTalons();
+    }
+    
+    private void initArmTalons() {
+        TalonFXConfiguration toApply = new TalonFXConfiguration();
+
+        toApply.MotorOutput.Inverted = InvertedValue.Clockwise_Positive;
+        toApply.CurrentLimits.SupplyCurrentLimit = ArmCal.ARM_SUPPLY_CURRENT_LIMIT_AMPS;
+        toApply.CurrentLimits.StatorCurrentLimit = ArmCal.ARM_STATOR_CURRENT_LIMIT_AMPS;
+        toApply.CurrentLimits.StatorCurrentLimitEnable = true;
+        toApply.MotorOutput.NeutralMode = NeutralModeValue.Coast;
+        toApply.Slot0.kP = ArmCal.ARM_MOTOR_P;
+        toApply.Slot0.kI = ArmCal.ARM_MOTOR_I;
+        toApply.Slot0.kD = ArmCal.ARM_MOTOR_D;
+        toApply.Slot0.kV = ArmCal.ARM_MOTOR_FF;
+
+        armMotorLeft.getConfigurator().apply(toApply);
+        armMotorRight.setControl(new Follower(armMotorLeft.getDeviceID(), true));
+    }
+
+    public void setDesiredPosition(ArmPosition armPosition) {
```

Comment:
James and I want to do interference zone control at the command level. ex. in scoringPrepSequence, we'll check if deep climb is in the way.

this method should just do this.armDesiredPosition = armPosition; because the commands will ensure movement is safe.